examples/realTimeScene.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python3
+
+import Sofa.Core
+import Sofa.Simulation
+import SofaRuntime
+import os
+import Sofa.splib.animation
+from time import time
+SofaRuntime.importPlugin("SofaOpenglVisual")
+
+_runAsPythonScript = False
+
+
+class RotationController(Sofa.Core.Controller):
+    """ This is a custom controller to perform actions when events are triggered """
+
+    def __init__(self, *args, **kwargs):
+        # These are needed (and the normal way to override from a python class)
+        Sofa.Core.Controller.__init__(self, *args, **kwargs)
+        print(" Python::__init__::" + str(self.name.value))
+
+        self.engine = kwargs["engine"]
+
+        self.inited = 0
+        self.iterations = 0
+        self.step = 1
+        self.max_iterations = 360
+        self.other_direction = False
+
+    def onAnimateBeginEvent(self, eventType):
+        self.iterations += 1
+        self.engine.rotation.value[0] += self.step if (
+            self.other_direction) else -self.step
+        if self.iterations > self.max_iterations:
+            self.iterations = 0
+            self.other_direction = not self.other_direction
+
+
+
+def createScene(root):
+
+    rpath =os.environ["SOFA_ROOT"]+"../src/share/mesh/"
+    root.dt = 0.5
+    root.name = 'root'
+    root.gravity = [0.0, 9.8, 0.0]
+
+    loader = root.addObject('MeshObjLoader', name='loader',
+                            filename=rpath + "liver.obj")
+    te = root.addObject(
+        "TransformEngine", name="te", input_position=loader.position.getLinkPath(), rotation=[0,0,0])
+    mo = root.addObject("MechanicalObject", name="mo",
+                        position=te.output_position.getLinkPath())
+
+    visu = root.addChild("Visu")
+    visu.addObject('OglModel', name="visu", src=loader.getLinkPath())
+    visu.addObject('IdentityMapping', name="BM", src=mo.getLinkPath())
+
+    Sofa.Simulation.init(root)
```

Comment:
Here is the part I found the most puzzling. 
The init/animate function of Simulation are supposed to be used within the main() function to control how the simulation is progressing. runSofa contains its own equivalent so normally in any case this kind of code should be out of createScene().

Reply:
This example was initially based on the ControllerScene.py example, which uses the init function this way. If i delete it, it appears the Animation Loop isn't initialized, which causes the scene to fail. I didn't know this function should be used only outside of createScene().